Plugins/Bitbucket/BitbucketPullRequestForm.cs

```diff
@@ -125,49 +116,57 @@ private List<PullRequest> GetPullRequests()
 
         private void BtnCreateClick(object sender, EventArgs e)
         {
-            if (ddlBranchSource.SelectedValue == null ||
-                ddlBranchTarget.SelectedValue == null ||
-                ddlRepositorySource.SelectedValue == null ||
-                ddlRepositoryTarget.SelectedValue == null)
-            {
-                return;
-            }
+            ThreadHelper.ThrowIfNotOnUIThread();
 
-            var info = new PullRequestInfo
-            {
-                Title = txtTitle.Text,
-                Description = txtDescription.Text,
-                SourceBranch = ddlBranchSource.SelectedValue.ToString(),
-                TargetBranch = ddlBranchTarget.SelectedValue.ToString(),
-                SourceRepo = (Repository)ddlRepositorySource.SelectedValue,
-                TargetRepo = (Repository)ddlRepositoryTarget.SelectedValue,
-                Reviewers = _reviewers
-            };
-            var pullRequest = new CreatePullRequestRequest(_settings, info);
-            var response = pullRequest.Send();
-            if (response.Success)
-            {
-                MessageBox.Show(_success.Text);
-                BitbucketViewPullRequestFormLoad(null, null);
-            }
-            else
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
             {
-                MessageBox.Show(string.Join(Environment.NewLine, response.Messages),
-                    _error.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
-            }
+                if (ddlBranchSource.SelectedValue == null ||
+                    ddlBranchTarget.SelectedValue == null ||
+                    ddlRepositorySource.SelectedValue == null ||
+                    ddlRepositoryTarget.SelectedValue == null)
+                {
+                    return;
+                }
+
+                var info = new PullRequestInfo
+                {
+                    Title = txtTitle.Text,
+                    Description = txtDescription.Text,
+                    SourceBranch = ddlBranchSource.SelectedValue.ToString(),
+                    TargetBranch = ddlBranchTarget.SelectedValue.ToString(),
+                    SourceRepo = (Repository)ddlRepositorySource.SelectedValue,
+                    TargetRepo = (Repository)ddlRepositoryTarget.SelectedValue,
+                    Reviewers = _reviewers
+                };
+                var pullRequest = new CreatePullRequestRequest(_settings, info);
+                var response = await pullRequest.SendAsync();
```

Comment:
❗️ Per #4751, a `SwitchToMainThreadAsync` needs to appear after this line.

Reply:
➡️ Added in 0c96467627f459b3c9adc0dadff2a919041d53e3

Reply:
Should it switch to a background thread before line:142?

Reply:
No. At line 142 it is an asynchronous call. At line 143 it is to ensure this control has not been disposed.

Reply:
Thanks. There's heaps for me to learn....